src/main/java/Duke.java

```diff
@@ -1,10 +1,232 @@
+import exception.DukeException;
+import exception.ErrorMessage;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.FileInputStream;
+import java.io.ObjectOutputStream;
+import java.io.ObjectInputStream;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+import java.util.ArrayList;
+
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
+    private static void logoWithHello() {
+        String logo = " ____        _\n"
+                + "|  _ \\ _   _| | _____\n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
     }
+
+    public static void horizontalLine() {
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void greet() {
+        horizontalLine();
+        System.out.println("Hello! I'm Duke\nWhat can I do for you?\n");
+        horizontalLine();
+    }
+
+    public static void exit() {
+        horizontalLine();
+        System.out.println("Bye. Hope to see you again soon!\n");
+        horizontalLine();
+    }
+
+    public static void processCommands(ArrayList<Task> tasks, String userInput) throws DukeException {
+        String[] words = userInput.split(" ");
+
+        if (words[0].equals("list")) {
+            listCommand(tasks);
+        } else if (words[0].equals("todo")) {
+            todoCommand(tasks, userInput);
+        } else if (words[0].equals("event")) {
+            eventCommand(tasks, userInput);
+        } else if (words[0].equals("deadline")) {
+            deadlineCommand(tasks, userInput);
+        } else if (words[0].equals("delete")) {
+            deleteCommand(tasks, words);
+        } else if (words[0].contains("mark")) {
+            markUnmarkCommand(tasks, words);
+        } else {
+            throw new DukeException(ErrorMessage.INVALID_COMMAND.toString());
+        }
+    }
+
+    private static void listCommand(ArrayList<Task> tasks) throws DukeException {
+        if (tasks.size() > 0) {
+            horizontalLine();
+            for (Task task : tasks) {
+                System.out.println((tasks.indexOf(task) + 1) + "." + task);
+            }
+            horizontalLine();
+        } else {
+            throw new DukeException(ErrorMessage.EMPTY_LIST.toString());
+        }
+    }
+
+    private static void todoCommand(ArrayList<Task> tasks, String userInput) throws DukeException {
```

Comment:
Please pay attention to the method names (as mentioned in the earlier comments by another student). 
E.g., it is better to name this `addTodo`

Alternatively, you could extract out the commands as individual classes, and have an `add` method in each class